### DISCUSSION TITLE
Asking about updateAudioSettings function

### DISCUSSION THREAD

**Discussion by saif-o99** [edumeet/edumeet-client#167] [Q&A] [answered]
https://github.com/edumeet/edumeet-client/blob/1bf4721fff69587cc7ac8e25f2f806a06ed67753/src/store/actions/mediaActions.tsx#L301

without passing `{restart:false}` to `updateMic` function it will recreate the media stream and all things. is this what we want here? or we want it to enter the `else if` that's inside `updateMic` and only update track constraints?

Another question. What's the whole idea behind using the `previewMicTrack` if found instead of creating a new media stream each time?

Thanks.

**Comment by saif-o99** [comment]
@havfo

**Comment by havfo** [comment, marked as answer]
This is intentional because of this: https://bugs.chromium.org/p/chromium/issues/detail?id=796964

We do this because we may have a track already from the join dialog.

**Comment by saif-o99** [comment]
> We do this because we may have a track already from the join dialog.

@havfo  Would it make a difference if we created a new stream each time instead of using the track from the join dialog?

**Reply by havfo** [reply]
The difference would be that it takes time to do a `getUserMedia`, so it is done to use the media we already have instead of waiting.

**Comment by saif-o99** [comment]
> This is intentional because of this: https://bugs.chromium.org/p/chromium/issues/detail?id=796964

`updateShareScreen` is different in this case? because `updateScreenshareSettings` doesn't send `{restart:true}` for `updateShareScreen`

**Reply by havfo** [reply]
Screenshare supports `applyConstraints`

**Comment by saif-o99** [comment]
https://github.com/edumeet/edumeet-client/blob/1bf4721fff69587cc7ac8e25f2f806a06ed67753/src/store/actions/mediaActions.tsx#L309C2-L309C2

Also, i think we should add `await` for `stopPreviewMic` because if its not finished , `updateMic` will see that there is a preview track with the old settings and will take it without creating new stream with the new settings

**Reply by havfo** [reply]
`stopPreviewMic` is async, but there is no await within it, because of this it will run synchronously. We should declare `stopPreviewMic` sync.

**Comment by saif-o99** [comment]
@havfo Many thanks and sorry for my too much questions

**Reply by havfo** [reply]
No worries :+1:
